lib/jsdom/living/helpers/create-element.js

```diff
@@ -0,0 +1,306 @@
+          throw new TypeError("Internal error: Invalid custom element.");
+        }
+
+        if (result._attributeList.length !== 0) {
+          throw DOMException.create(_globalObject, ["Unexpected attributes.", "NotSupportedError"]);
+        }
+        if (domSymbolTree.hasChildren(result)) {
+          throw DOMException.create(_globalObject, ["Unexpected child nodes.", "NotSupportedError"]);
+        }
+        if (domSymbolTree.parent(result)) {
+          throw DOMException.create(_globalObject, ["Unexpected element parent.", "NotSupportedError"]);
+        }
+        if (result._ownerDocument !== document) {
+          throw DOMException.create(_globalObject, ["Unexpected element owner document.", "NotSupportedError"]);
+        }
+        if (result._namespaceURI !== namespace) {
+          throw DOMException.create(_globalObject, ["Unexpected element namespace URI.", "NotSupportedError"]);
+        }
+        if (result._localName !== localName) {
+          throw DOMException.create(_globalObject, ["Unexpected element local name.", "NotSupportedError"]);
+        }
+
+        result._prefix = prefix;
+        result._isValue = isValue;
+      } catch (error) {
+        reportException(document._defaultView, error);
+
+        const interfaceWrapper = interfaces.getInterfaceWrapper("HTMLUnknownElement");
+        result = interfaceWrapper.createImpl(_globalObject, [], {
+          ownerDocument: document,
+          localName,
+          namespace: HTML_NS,
+          prefix,
+          ceState: "failed",
+          ceDefinition: null,
+          isValue: null
+        });
+      }
+    } else {
+      const interfaceWrapper = interfaces.getInterfaceWrapper("HTMLElement");
+      result = interfaceWrapper.createImpl(_globalObject, [], {
+        ownerDocument: document,
+        localName,
+        namespace: HTML_NS,
+        prefix,
+        ceState: "undefined",
+        ceDefinition: null,
+        isValue: null
+      });
+
+      enqueueCEUpgradeReaction(result, definition);
+    }
+  } else {
+    let elementInterface;
+
+    switch (namespace) {
+      case HTML_NS:
+        elementInterface = getHTMLElementInterface(localName);
+        break;
+
+      case SVG_NS:
+        elementInterface = getSVGInterface(localName);
+        break;
+
+      default:
+        elementInterface = interfaces.getInterfaceWrapper("Element");
+        break;
+    }
+
+    result = elementInterface.createImpl(_globalObject, [], {
+      ownerDocument: document,
+      localName,
+      namespace,
+      prefix,
+      ceState: "uncustomized",
+      ceDefinition: null,
+      isValue
+    });
+
+    if (namespace === HTML_NS && (isValidCustomElementName(localName) || isValue !== null)) {
+      result._ceState = "undefined";
+    }
+  }
+
+  return result;
+}
+
+// https://dom.spec.whatwg.org/#internal-createelementns-steps
+function createElementNS(document, namespace, qualifiedName, options) {
```

Comment:
Could we call this `internalCreateElementNSSteps`? Right now it seems too much like a peer of `createElement` where in the spec "create an element" is a lot more fundamental.